applications/vanilla/models/class.discussionmodel.php

```diff
@@ -1284,9 +1289,9 @@ public function getByUser($userID, $limit, $offset, $lastDiscussionID = false, $
             return new Gdn_DataSet([]);
         }
 
-        // Allow us to set perspective of a different user.
-        if (empty($watchUserID)) {
-            $watchUserID = $userID;
+        // If no user was provided, view from the perspective of a guest.
+        if (!filter_var($userID, FILTER_VALIDATE_INT)) {
+            $watchUserID = UserModel::GUEST_USER_ID;
```

Comment:
Not sure about new code equal to previous one.
Previous check was done against `$watchUserID` and set new value of it only if it was empty.
New code checks another variable `$userID` and set `$watchUserID` only if `$userID` is not valid integer.
hmmmm... Not clear refactoring to me.